### PR TITLE
boards: m5stack: cores3: Add FT6336 configuration

### PR DESCRIPTION
**Pull request by soburi**
Adding FT6336 touchpanel controller configuration.

### CODE REVIEW

**Comment by soburi** [comment]
@sylvioalves @marekmatej

Thank you for advance your cooperation with this as well.

File: boards/m5stack/m5stack_cores3/Kconfig.m5stack_cores3 [hunk 1]
```diff
@@ -8,3 +8,13 @@ config BOARD_M5STACK_CORES3
 	select SOC_ESP32S3_PROCPU if BOARD_M5STACK_CORES3_ESP32S3_PROCPU || \
 				     BOARD_M5STACK_CORES3_ESP32S3_PROCPU_SE
 	select SOC_ESP32S3_APPCPU if BOARD_M5STACK_CORES3_ESP32S3_APPCPU
+
+if SOC_ESP32S3_PROCPU
```
Comment (sylvioalves) [inline boards/m5stack/m5stack_cores3/Kconfig.m5stack_cores3:12]: Please, move this to `Kconfig.defconfig` file:

```c
if BOARD_M5STACK_CORES3_ESP32S3_PROCPU

config INPUT_FT5336_INTERRUPT
	default y if INPUT

config INPUT
	default y

endif # BOARD_M5STACK_CORES3_ESP32S3_PROCPU
```

Comment (soburi) [reply boards/m5stack/m5stack_cores3/Kconfig.m5stack_cores3]: Thank you for your comment.
I applied your suggestion and added modifications to make it compatible with variants.

File: boards/m5stack/m5stack_cores3/m5stack_cores3_procpu_common.dtsi [hunk 2]
```diff
@@ -19,6 +19,7 @@
 		zephyr,code-partition = &slot0_partition;
 		zephyr,rtc = &bm8563_rtc;
 		zephyr,bt-hci = &esp32_bt_hci;
+		zephyr,touch = &ft6336_touch;
```
Comment (marekmatej) [inline boards/m5stack/m5stack_cores3/m5stack_cores3_procpu_common.dtsi:22]: `ft6336` looks to me like a typo. Driver is called `ft5336`.

File: boards/m5stack/m5stack_cores3/m5stack_cores3_procpu_common.dtsi [hunk 3]
```diff
@@ -30,6 +31,11 @@
 		watchdog0 = &wdt0;
 		rtc = &bm8563_rtc;
 	};
+
+	lvgl_pointer {
+		compatible = "zephyr,lvgl-pointer-input";
+		input = <&ft6336_touch>;
```
Comment (marekmatej) [inline boards/m5stack/m5stack_cores3/m5stack_cores3_procpu_common.dtsi:37]: here

**Review by marekmatej — CHANGES_REQUESTED** [review]
Except the typo LGTM

File: boards/m5stack/m5stack_cores3/m5stack_cores3_procpu_common.dtsi (hunk 2 above)
Comment (sylvioalves) [reply boards/m5stack/m5stack_cores3/m5stack_cores3_procpu_common.dtsi:22]: It is not a typo. FT6336 is a variant and can be used with ft5663 driver:
```
description: FT3267/FT5XX6/FT6XX6 capacitive touch panels

compatible: "focaltech,ft5336"
```

Comment (marekmatej) [reply boards/m5stack/m5stack_cores3/m5stack_cores3_procpu_common.dtsi:22]: ok, I didn't know that. Thx for clarifying this!

**Review by marekmatej — APPROVED** [review]

**Comment by soburi** [comment]
@sylvioalves 
Could you take a look if you have a while?

**Review by sylvioalves — APPROVED** [review]

**Pull request merged**